gz-resources.html

```diff
@@ -92,126 +121,193 @@
+        })
+        socket.on('reconnect_error',  function(err) {
+          console.error(msg, 'socket reconnect error ' + util.inspect(err))
+        })
+        socket.on('resource', function(resource) {
+          console.log(resource)
+          that.fire('resource', resource)
+        })
+        return this.socket
+      },
+
+      /// disconnect the websocket
+      disconnect: function() {
+        if (this.socket) {
+          try {
+            this.socket.disconnect()
+          }
+          catch (err) {
+            console.error('<gz-resources> disconnect', err)
+          }
+        }
       },
 
-      //////////////////////////////////////////////////
       /// Get all
-      //////////////////////////////////////////////////
       /// Get all resources for user
       getAll: function(req) {
         this.$.ajaxgetall.headers.authorization = this.token.trim()
         this.$.ajaxgetall.generateRequest()
+        console.log(this.$.ajaxgetall.method,this.$.ajaxgetall.url, 'request sent')
       },
 
       /// Callback when successfully retrieving all resources
-      _getAllResponse: function(req) {
+      _getAllResponse: function() {
         const resp = this.$.ajaxgetall.lastResponse
+        console.log('_getAllResponse:', resp)
         if (!resp)
           return;
-
-        if (!resp.success)
-        {
-          // ERROR
-          return;
-        }
-
-        this.resources = resp.result;
-
+        this.resources = resp.result
         this.fire('resourcesupdated', this.resources)
       },
 
       /// Callback when failed to retrieve all resources
       _getAllError: function(err) {
-        console.error(err)
+        this._errorHelper(err, 'Failed to get ' + this.resourcename + 's')
       },
 
-      //////////////////////////////////////////////////
       /// Create resource
-      //////////////////////////////////////////////////
-      /// Tap resource delete buttom on config
-      _tapCreate: function(req) {
+      /// Tap resource delete button on config
+      _tapCreate: function() {
         try {
-          this.data = JSON.parse(this.datatxt);
+          this.data = JSON.parse(this.datatxt)
         }
         catch(e) {
-          console.error("Couldn't parse json: " + e);
-          return;
+          const msg = "Create resource error: Couldn't parse json: " + e
+          this.fire('error', msg,  {bubbles: this.bubbles})
+          return
         }
-        this.create();
+        this.$.ajaxcreate.headers.authorization = this.token.trim()
+        this.$.ajaxcreate.body = this.data
+        this.$.ajaxcreate.generateRequest()
       },
 
-      /// Create a resource
-      create: function(req) {
-        this.$.ajaxcreate.headers.authorization = this.token.trim();
-        this.$.ajaxcreate.body = this.data;
-        this.$.ajaxcreate.generateRequest();
+      /// Create a resource from data
+      create: function(data) {
+        // set data
+        this.data = data
```

Comment:
Sorry for the bug. I fixed this with a common function called from either functions